aioshelly/rpc_device/device.py

```diff
@@ -568,10 +568,15 @@ async def _retrieve_blutrv_components(self, components: dict[str, Any]) -> None:
 
             cfg: dict[str, Any] = results[0]["config"]["trv:0"]
             # addr, name and model_id must be added from Shelly.GetComponents call
+            _attrs = component.get("attrs")
             cfg.update({"addr": component["config"]["addr"]})
             cfg.update({"name": component["config"]["name"]})
             cfg.update(
-                {"local_name": BLU_TRV_MODEL_ID.get(component["attrs"]["model_id"])}
+                {
+                    "local_name": BLU_TRV_MODEL_ID.get(_attrs.get("model_id"))
+                    if _attrs
+                    else None
+                }
```

Comment:
If we set the default to `{}` we don't need this condition

Reply:
Good catch !